File: Protótipos(back-end)/projetoFila(ts)/Fila.ts
import Cliente from "./Cliente";

class Fila {
    private lugares: number;
    private clientes: Cliente[];

    constructor(lugares: number) {
        this.lugares = lugares;
        this.clientes = [];
    }

    adicionarCliente(cliente: Cliente): void {
        this.clientes.push(cliente);
    }

    getClientes(): Cliente[] {
        return this.clientes;
    }

    getLugares(): number {
        return this.lugares;
    }
}

export default Fila;

File: Protótipos(back-end)/projetoFila(java)/Fila.java
package projetoMascate;

import java.util.ArrayList;
import java.util.List;

public class Fila {
    private int lugares;
    private List<Cliente> clientes;

    public Fila(int lugares) {
        this.lugares = lugares;
        this.clientes = new ArrayList<>();
    }

    public void adicionarCliente(Cliente cliente) {
        clientes.add(cliente);
    }

    public List<Cliente> getClientes() {
        return clientes;
    }

    public int getLugares() {
        return lugares;
    }
}
